apps/antalmanac/src/components/RightPane/CoursePane/SearchForm/RestrictionsFilter.tsx

```diff
@@ -0,0 +1,183 @@
+    [String('R')]: 'R: Biomedical Pass/Fail course (School of Medicine only)',
+    [String('K')]: 'K: Graduate only',
+    [String('S')]: 'S: Satisfactory/Unsatisfactory only',
+    [String('B')]: 'B: Authorization code required',
+    [String('L')]: 'L: Major only',
+};
+
+interface RestrictionFilterProps {
+    classes: ClassNameMap;
+}
+
+interface RestrictionFilterState {
+    restrictions: string[];
+}
+
+// url is not being grabbed to set the values correctly on refresh
+class RestrictionsFilter extends PureComponent<RestrictionFilterProps, RestrictionFilterState> {
+    updateRestrictionsAndGetFormData() {
+        RightPaneStore.updateFormValue('restrictions', RightPaneStore.getUrlRestrictionsValue());
+        return RightPaneStore.getFormData().restrictions;
+    }
+
+    getRestrictions() {
+        return RightPaneStore.getUrlRestrictionsValue().trim()
+            ? this.updateRestrictionsAndGetFormData()
+            : RightPaneStore.getFormData().restrictions;
+    }
+
+    state = {
+        restrictions:
+            this.getRestrictions() !== 'ALL' && typeof this.getRestrictions() === 'string' // guards for type errors
+                ? this.getRestrictions()
+                      .split('')
+                      .filter((item: any) => !Array.isArray(item))
+                      .map((restriction) => staticRestrictionList[restriction as string]) // Converts URL letters to the full code so the Select / Checkboxes can be checked/unchecked
+                : [this.getRestrictions()],
+    };
+
+    handleChange = (event: ChangeEvent<{ restrictions?: string | undefined; value: unknown }>) => {
+        let value: unknown;
+
+        if ((event.target.value as string[])[0] == 'ALL' || Array.isArray((event.target.value as string[])[0])) {
+            value = (event.target.value as string[]).slice(1);
+        } else {
+            value = event.target.value as string[];
+        }
+
+        this.setState(
+            {
+                restrictions: value as string[],
+            },
+            () => {
+                RightPaneStore.updateFormValue('restrictions', value as unknown as string);
+            }
+        );
+
+        const stateObj = { url: 'url' };
+        const url = new URL(window.location.href);
+        const urlParam = new URLSearchParams(url.search);
+        urlParam.delete('restrictions');
+
+        if ((value as string[]) && (value as string[])[0] !== undefined) {
+            urlParam.append(
+                'restrictions',
+                (value as string[])
+                    .map((value) => value.split(':')[0].trim()) // Prevents the URL from becoming too long
+                    .sort((a, b) => a.localeCompare(b))
+                    .join('')
+            );
+        }
+
+        const param = urlParam.toString();
+        const new_url = `${param.trim() ? '?' : ''}${param}`;
+        history.replaceState(stateObj, 'url', '/' + new_url);
```

Comment:
We should generally avoid manipulating the browser history directly because we're using client side routing with React-Router-Dom

Reply:
90% of what's here is ripped from `GESelector.tsx` (and by extension, everything else under `SearchForm), so I'm gonna leave as is and call it Tech Debt 🤓